bridgepoint/prebuild.py

```diff
@@ -1852,19 +1892,22 @@ def prebuild_action(instance):
     - SPR_PS
     '''
     walker_map = {
-        'S_SYNC': FunctionPrebuilder,
-        'S_BRG': BridgePrebuilder,
-        'O_TFR': OperationPrebuilder,
-        'O_DBATTR': DerivedAttributePrebuilder,
-        'SM_ACT': TransitionPrebuilder,
-        'SPR_RO': RequiredOperationPrebuilder,
-        'SPR_RS': RequiredSignalPrebuilder,
-        'SPR_PO': ProvidedOperationPrebuilder,
-        'SPR_PS': ProvidedSignalPrebuilder
+        'S_SYNC': {'prebuilder': FunctionPrebuilder, 'elementName': 'Function'},
+        'S_BRG': {'prebuilder': BridgePrebuilder, 'elementName': 'Bridge'},
+        'O_TFR': {'prebuilder': OperationPrebuilder, 'elementName': 'Class Operation'},
+        'O_DBATTR': {'prebuilder': DerivedAttributePrebuilder, 'elementName': 'Derived Base Attribute'},
+        'SM_ACT': {'prebuilder': TransitionPrebuilder, 'elementName': 'State/Transaction'},
+        'SPR_RO': {'prebuilder': RequiredOperationPrebuilder, 'elementName': 'Required Operation'},
+        'SPR_RS': {'prebuilder': RequiredSignalPrebuilder, 'elementName': 'Required Signal'},
+        'SPR_PO': {'prebuilder': ProvidedOperationPrebuilder, 'elementName': 'Provided Operation'},
+        'SPR_PS': {'prebuilder': ProvidedSignalPrebuilder, 'elementName': 'Provided Signal'}
     }
     metaclass = xtuml.get_metaclass(instance)
-    walker = walker_map[metaclass.kind](metaclass.metamodel, instance)
-    logger.info('processing action %s' % walker.label)
+    walker = walker_map[metaclass.kind]['prebuilder'](
+        metaclass.metamodel, instance)
+    actionType = walker_map[metaclass.kind]['elementName']
```

Comment:
How about moving the action_type as a property to each Prebuilder class, e.g.,

FunctionPrebuilder(ActionPrebuilder):
    element_name = 'Function'

*